### PR TITLE
tests: net: Application for automatic network testing

### PR DESCRIPTION
**Pull request by jukkar**
This is a simple network application that is used by automatic
testing scripts. The application currently enables IPv6 and IPv4,
TCP and UDP, VLAN and traffic class support, and echo-server service.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

**Comment by jukkar** [comment]
> Which automatic testing scripts? Can the information be added to the commit message?

This is still very much WIP and is related to https://github.com/intel/net-test-suites project by @ozhuraki

**Comment by pfalcon** [comment]
> related to https://github.com/intel/net-test-suites project by @ozhuraki

Thanks, I wondered whether there's any update on TTCN stuff.

File: tests/net/automatic_testing/README.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+************
+
+- :ref:`networking_with_qemu`
+
+Building and Running
+********************
+
+Normally this test application is launched by automatic test system.
+It is also possible to run this testing application with QEMU as described in
+:ref:`networking_with_qemu`, with native-posix board or with real hardware.
```
Comment (dbkinder) [inline tests/net/automatic_testing/README.rst]: ``board, or with``  (add a comma)

File: tests/net/automatic_testing/README.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+Follow these steps to build the testing application:
+
+.. zephyr-app-commands::
+   :zephyr-app: tests/net/automatic_testing
+   :board: <board to use>
+   :conf: prj.conf
+   :goals: build
+   :compact:
+
+If this application is run in native_posix board, then normally one needs extra
```
Comment (dbkinder) [inline tests/net/automatic_testing/README.rst]: change ``normally one needs`` to ``you will need``

File: tests/net/automatic_testing/README.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+priviliges to create and configure the TAP device in the host system.
+Use sudo command to execute the Zephyr process with admin privileges, like this:
```
Comment (dbkinder) [inline tests/net/automatic_testing/README.rst]: Use the ``sudo`` command to execute the Zephyr process with admin privileges:

File: tests/net/automatic_testing/README.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+- VLAN tag 100: IPv4 198.51.100.1 and IPv6 2001:db8:100::1
+- VLAN tag 200: IPv4 203.0.113.1 and IPv6 2001:db8:200::1
+- IPv4 192.0.2.1 and IPv6 2001:db8::1
+
+Setting up Linux Host
+=====================
+
+The :file:`samples/net/vlan/vlan-setup-linux.sh` provides a script that can be
+executed on the Linux host. It creates two VLAN on the Linux host and
+suitable IP routes to Zephyr. This script is not needed for native-posix board
```
Comment (dbkinder) [inline tests/net/automatic_testing/README.rst]: `` for a native-posix board as the host IP address...``  (adding ``a`` and ``the``)

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by jukkar** [comment]
Fixed the documentation.

**Review by dbkinder — APPROVED** [review]
Thanks!

**Comment by pfalcon** [comment]
As we with @tbursztyka finally gave a run to @ozhuraki's https://github.com/intel/net-test-suites at the recent sprint, I wonder what's the status of this? Currently, net-test-suites runs against echo_server, I wonder if this can enable more coverage (but how soon, etc.?)

**Comment by jukkar** [comment]
This is currently not progressing so closing it for time being. Can be re-opened later if needed.

**Pull request closed without merging**